protobuf.sh

```diff
@@ -11,6 +11,12 @@ prefer_system_check: |
 
 rsync -av --delete --exclude="**/.git" $SOURCEDIR/ .
 autoreconf -ivf
+
+# Set the environment variables CC and CXX if a compiler is defined in the defaults file 
+# In case CC and CXX are defined the corresponding compilers are used during compilation  
+[[ -z "$CXX_COMPILER" ]] || export CXX=$CXX_COMPILER
```

Comment:
Why do you need the export? If it's defined in the defaults environment it should be always exported already. Is that not the case?

Reply:
Because CXX is not defined in the defaults. I tried to change the recipes in such a way that they still work with default files which does not define the compiler. This should allow to gradually change the recipes. Only if all recipes support the new scheme one would change the default files. For the compilation of O2 all recipes support an external compiler, but I did not touch any recipe which is not needed for the 02 recipe.

That's why I defined variables like CXX_COMPILER in the defaults file which are then used to setup the needed variables like CXX. If there is a possibility to do so in the defaults file this would be the cleaner solution.